### PR TITLE
ActiveFedora::File.mime_type should be updatable

### PR DESCRIPTION
**Pull request by cjcolvar**
Squash and rebase of file_mime_types branch by @tpendragon

### CODE REVIEW

**Comment by jcoyne** [comment]
@cjcolvar It looks like rubocop found some issues with your changes.  Can you run `bundle exec rubocop -a` on the code and commit the changes?

File: lib/active_fedora/with_metadata/metadata_node.rb [hunk 1]
```diff
@@ -45,7 +46,7 @@ def save
 
       def changed_attributes
         super.tap do |changed|
-          changed['type'] = true if type.present?
+          changed['type'] = true if type.present && new_record?
```
Comment (jcoyne) [inline lib/active_fedora/with_metadata/metadata_node.rb]: It looks like the method is missing a question mark `present?`

**Comment by cjcolvar** [comment]
Thanks @jcoyne.  Everything is green now.

**Comment by jcoyne** [comment]
@cjcolvar I'm going to revert this commit, because it puts FCrepo 4.5.0 into a bad state. See https://github.com/projecthydra-labs/curation_concerns/pull/693

**Pull request merged**